src/debugAdapter/goDebug.ts

```diff
@@ -246,7 +263,22 @@ class Delve {
 	}
 
 	close() {
-		this.debugProcess.kill();
+        if (this.debugProcess === null) {                                                    
+            this.call<DebuggerState>('Command', [{ name: 'halt' }], (err, state) => {
+                if (err) {
+                    console.error('Failed to halt.');
+                }
+                // else TODO:  Can't find a way to kill remote delve.
+                // {
+                //     this.call<number>('Detach', [true], (err, ret) => {
```

Comment:
I removed the commented out code.  I think the other code should remain though.  I shut down remote dlv process by hitting Ctrl-C.  If I don't issue halt command, the remote dlv process will not shut down gracefully.  It will hang trying to delete the debug binary and you will have to hit Ctrl-X and process will be left open.  With halt command, hitting Ctrl-C will shut down process smoothly.